python/pyspark/sql/functions.py

```diff
@@ -22,20 +22,10 @@
 import sys
 import functools
 import warnings
-from typing import (
-    Any,
-    cast,
```

Comment:
How about just fix the `cast` instead of fixing all imports and add a test to verify it ??

IMHO most of the other functions except `cast` seem to be names that are very unlikely to be used repeatedly.

Another concern is that we're using the typing module most of our PySpark files this way, so I don't know maybe if we need to fix all the other files for consistency(but which seems to be too much work).

```shell
spark$ git grep "from typing import" python/pyspark | wc -l
     202
```

Reply:
I agree that the change seemed quite cumbersome. It made me wish Python had some kind of "private import" keyword to handle such cases more easily.

I agree with you that `cast` is the only name that might be confusing (perhaps `overload` too, but all the other names start with an uppercase). The `functions` module feels a little special to me because it is the module I use the most as a Spark user, it's definitely a public API. The 201 other modules don't require such change.

It's out of the scope of this MR, but perhaps for the long term you could consider reorganizing this module [the same way as I did in one of my projects](https://github.com/FurcyPin/spark-frame/blob/main/spark_frame/transformations.py), which had two advantages:
- the code of each method was isolated in a separate file (that would prevent having a 10 000-lines files)
- there was no import pollution

For now, I'll do as you suggest: only handle `typing.cast` as a special case and add a unit test to make sure it does not get imported again.

Reply:
In the end, I went for `from typing import cast as _cast` which makes the intent even more explicit, I think.